sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -257,7 +257,7 @@ mod checked {
             metrics,
             move_vm,
             &mut temporary_store,
-            tx_context,
+            tx_context.clone(),
```

Comment:
to change this API to have a `TXContext` and not a `&mut` requires a change in the executor API (so all version) and I am wondering whether we should do it or just let it go.
Thoughts?

Reply:
Why do we need an API change here?

Reply:
this API used to take a `&mut TxContext` (boundary between core and execution) though here we take the `TxContext` by value. So we clone the mutable reference in order to pass it by value, which is kind of lame as we could have passed the context by value in the first place

Reply:
though there may be issues the way the code is changed now, looking into it and we'll talk more

Reply:
well it turns out the "fix" I put breaks a genesis test exactly because it's updating the context in what seems an incompatible way.
Now I am really confused

Reply:
ok we'll talk later about this - lol